contracts/LiquidationsPeripheral.vy

```diff
@@ -710,118 +781,105 @@ def payLoanLiquidationsGracePeriod(_loanId: uint256, _erc20TokenContract: addres
     assert block.timestamp > liquidation.gracePeriodMaturity, "liquidation in grace period"
     assert block.timestamp <= liquidation.lenderPeriodMaturity, "liquidation out of lender period"
-
     assert ILendingPoolPeripheral(self.lendingPoolPeripheralAddresses[liquidation.erc20TokenContract]).lenderFunds(msg.sender).currentAmountDeposited > 0, "msg.sender is not a lender"
 
-    ILiquidationsCore(self.liquidationsCoreAddress).removeLiquidation(_collateralAddress, _tokenId)
-
-    log LiquidationRemoved(
-        liquidation.erc20TokenContract,
-        liquidation.collateralAddress,
-        liquidation.lid,
-        liquidation.collateralAddress,
-        liquidation.tokenId,
-        liquidation.erc20TokenContract,
-        liquidation.loansCoreContract,
-        liquidation.loanId,
-        liquidation.borrower
-    )
-
-    fundsSender: address = msg.sender
     lenderPeriodInterestAmount: uint256 = self._computeLiquidationInterestAmount(
         liquidation.principal,
         liquidation.interestAmount
     )
-    if liquidation.lenderPeriodPrice > liquidation.principal + lenderPeriodInterestAmount:
-        fundsSender = self
-        IERC20(liquidation.erc20TokenContract).transferFrom(
+    _liquidation_amount: uint256 = liquidation.principal + lenderPeriodInterestAmount
+
+    lendingPoolPeripheral: address = self.lendingPoolPeripheralAddresses[liquidation.erc20TokenContract]
+    
+    receivedAmount: uint256 = msg.value
+    ethPayment: bool = receivedAmount > 0
+    if ethPayment:
+        assert receivedAmount >= _liquidation_amount, "insufficient value received"
+        log PaymentReceived(msg.sender, msg.sender, receivedAmount)
+        ILendingPoolPeripheral(lendingPoolPeripheral).receiveFundsFromLiquidationEth(
             msg.sender,
-            self,
-            liquidation.lenderPeriodPrice
-        )
-        IERC20(liquidation.erc20TokenContract).approve(
-            ILendingPoolPeripheral(self.lendingPoolPeripheralAddresses[liquidation.erc20TokenContract]).lendingPoolCoreContract(),
-            liquidation.principal + lenderPeriodInterestAmount
+            liquidation.principal,
+            lenderPeriodInterestAmount,
+            True,
+            liquidation.principal,
+            "liquidation_lenders_period",
+            value=_liquidation_amount
         )
-        IERC20(liquidation.erc20TokenContract).transfer(
-            liquidation.borrower,
-            liquidation.lenderPeriodPrice - (liquidation.principal + lenderPeriodInterestAmount)
+        log PaymentSent(lendingPoolPeripheral, lendingPoolPeripheral, _liquidation_amount)
+    else:
+        ILendingPoolPeripheral(lendingPoolPeripheral).receiveFundsFromLiquidationWeth(
+            msg.sender,
+            liquidation.principal,
+            lenderPeriodInterestAmount,
+            True,
+            liquidation.principal,
+            "liquidation_lenders_period"
```

Comment:
Before these changes, there was logic here that would do the following:
* if `liquidation.lenderPeriodPrice > liquidation.principal + lenderPeriodInterestAmount`, the excess would be sent to the borrower

As far as I can see, we are disregarding that logic entirely and are potentially losing value in the liquidation in the scenario of the condition above, i.e. we're neither sending the excess to the borrower nor are we letting the lenders (and us) keep it because we're always using the `__liquidation_amount` (or its components individually).

As of now, and to keep consistent with the logic in `liquidateNFTX`, we should keep the `liquidation.lenderPeriodPrice` for the lenders and Zharta, regardless of its value. This means that we can let go of `_liquidation_amount` and set the `lenderPeriodInterestAmount = liquidation.lenderPeriodPrice - liquidation.principal`

Was it clear or did I understand the code in the wrong way? 😛